omod/src/main/java/org/openmrs/module/pihcore/fragment/controller/dashboardwidgets/PregnancyIndicatorsFragmentController.java

```diff
@@ -0,0 +1,157 @@
+        ));
+
+        question = conceptService.getConceptByUuid(SierraLeoneConfigConstants.CONCEPT_DATETIMEOFBIRTH_UUID);
+        if ( question != null ) {
+            obsWithinProgram = PihCoreUtils.getObsWithinProgram(patient, new HashSet<>(Arrays.asList(question)), null, null);
+        }
+        fields.put("latestDelivery", SimpleObject.create(
+                "label", "pihcore.latest.delivery",
+                "obs", (obsWithinProgram!=null && obsWithinProgram.size() > 0 ) ? obsWithinProgram.get(0) : ""
+        ));
+        question = conceptService.getConceptByUuid(PihEmrConfigConstants.CONCEPT_TYPEOFDELIVERY_UUID);
+        if ( question != null ) {
+            obsWithinProgram = PihCoreUtils.getObsWithinProgram(patient, new HashSet<>(Arrays.asList(question)), null, null);
+        }
+        fields.put("deliveryType", SimpleObject.create(
+                "label", "pihcore.latest.delivery.type",
+                "obs", (obsWithinProgram!=null && obsWithinProgram.size() > 0 ) ? obsWithinProgram.get(0) : ""
+        ));
+
+        EncounterSearchCriteriaBuilder searchCriteriaBuilder = new EncounterSearchCriteriaBuilder();
+        searchCriteriaBuilder.setPatient(patient);
+        searchCriteriaBuilder.setEncounterTypes(new HashSet<>(Arrays.asList(emrApiProperties.getAdmissionEncounterType())));
+        List<Encounter> encounters = encounterService.getEncounters(searchCriteriaBuilder.createEncounterSearchCriteria());
+        Date lastAdmissionDatetime = null;
+        for (Encounter encounter : encounters) {
+            Date encounterDatetime = encounter.getEncounterDatetime();
+            if ( lastAdmissionDatetime == null || lastAdmissionDatetime.before(encounterDatetime)) {
+                // find the most recent Admission datetime
+                lastAdmissionDatetime = encounterDatetime;
+            }
+        }
+        fields.put("latestAdmission", SimpleObject.create(
+                "label", "pihcore.latest.admission",
+                "obs",  lastAdmissionDatetime != null ? lastAdmissionDatetime : ""
+        ));
+
+        Set<EncounterType> ancEncTypes = new HashSet<>();
+        ancEncTypes.add(encounterService.getEncounterTypeByUuid(PihEmrConfigConstants.ENCOUNTERTYPE_ANC_INTAKE_UUID));
+        ancEncTypes.add(encounterService.getEncounterTypeByUuid(PihEmrConfigConstants.ENCOUNTERTYPE_ANC_FOLLOWUP_UUID));
+
+        searchCriteriaBuilder = new EncounterSearchCriteriaBuilder();
+        searchCriteriaBuilder.setPatient(patient);
+        searchCriteriaBuilder.setEncounterTypes(ancEncTypes);
+        if ( programUuid != null ) {
+            searchCriteriaBuilder.setFromDate(PihCoreUtils.getProgramEnrollmentDate( patient, programUuid ));
+        }
+        encounters = encounterService.getEncounters(searchCriteriaBuilder.createEncounterSearchCriteria());
+        int numberOfVisits = encounters != null ? encounters.size() : 0;
```

Comment:
Looking at the specifications, this follows from that, but wouldn't we want to collapse this by visits, ie the number of visits with an ANC Intake or ANC Followup encounter?  Granted, a visit should only have one, but this would prevent against over-counting in the case of duplicate form entry?

Reply:
Sorry, I did not understand your question here. We discussed this yesterday during the standup, and we could discuss more tomorrow?